### PR TITLE
initial commit aliceclonergb

### PR DESCRIPTION
**Pull request by SneakboxKB**
Initial commit for AliceClone RGB

## Types of Changes

- [X] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [X] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [X] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [X] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [X] I have updated the documentation accordingly.
- [X] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [X] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/sneakbox/aliceclonergb/keymaps/via/keymap.c [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+        }
+    }
+    return true;
+}
+
+
+//function for layer indicator LED
+uint32_t layer_state_set_user(uint32_t state)
+{
+    if (biton32(state) == 1) {
```
Comment (zvecr) [inline keyboards/sneakbox/aliceclonergb/keymaps/via/keymap.c:77]:
```suggestion
    if (get_highest_layer(state) == 1) {
```

File: keyboards/sneakbox/aliceclonergb/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        } else {
+            tap_code(KC_UP);
+        }
+    }
+    return true;
+}
+
+
+//function for layer indicator LED
+uint32_t layer_state_set_user(uint32_t state)
```
Comment (zvecr) [inline keyboards/sneakbox/aliceclonergb/keymaps/via/keymap.c:75]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state)
```

File: keyboards/sneakbox/aliceclonergb/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+  } else {
+    writePinLow(D7);
+  }
+  if (IS_LED_ON(usb_led, USB_LED_CAPS_LOCK)) {
+    writePinHigh(D6);
+  } else {
+    writePinLow(D6);
+  }
+
+}
```
Comment (zvecr) [inline keyboards/sneakbox/aliceclonergb/keymaps/via/keymap.c:98]:
```suggestion
bool led_update_user(led_t led_state) {
  if (led_state.num_lock) {
    writePinHigh(D7);
  } else {
    writePinLow(D7);
  }
  if (led_state.caps_lock) {
    writePinHigh(D6);
  } else {
    writePinLow(D6);
  }
  return false;
}
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Review by fauxpark — APPROVED** [review]

File: keyboards/sneakbox/aliceclonergb/readme.md [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+
+![Sneakbox Alice Clone RGB PCB](https://i.imgur.com/pMZNqLXl.jpg)
+
+An Alice-compatible PCB with hotswap with pushbutton rotary encoder socket placed at the top left key position. This PCB is the successor to the v3 AliceClone PCB and features RGB underglow lighting along with revised hardware components.
+
+* Keyboard Maintainer: [mujimanic](https://github.com/mujimanic)
+* Hardware Supported: Any Alice-layout case.
+* Hardware Availability: [sneakbox.design](https://sneakbox.design/products/sneakbox-alice-hotswap-pcbs)
+
+Make example for this keyboard (after setting up your build environment):
```
Comment (drashna) [inline keyboards/sneakbox/aliceclonergb/readme.md:11]: Could you add instructions on how to enter the bootloader, to flash new firmware? 

Eg: 
```markup
## Bootloader

Enter the bootloader in 3 ways:

* **Bootmagic reset**: Hold down the key at (0,0) in the matrix (usually the top left key or Escape) and plug in the keyboard
* **Physical reset button**: Briefly press the button on the back of the PCB - some may have pads you must short instead
* **Keycode in layout**: Press the key mapped to `RESET` if it is available
```

File: keyboards/sneakbox/aliceclonergb/rules.mk [hunk 5]
```diff
@@ -0,0 +1,24 @@
+# MCU name
+MCU = atmega32u4
+
+# Bootloader selection
+BOOTLOADER = atmel-dfu
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = lite     # Enable Bootmagic Lite
```
Comment (drashna) [inline keyboards/sneakbox/aliceclonergb/rules.mk:10]: There is no "full" bootmagic anymore, only the "lite" option, which can be turned on with just "yes".
```suggestion
BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
```

File: keyboards/sneakbox/aliceclonergb/rules.mk [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
```
Comment (drashna) [inline keyboards/sneakbox/aliceclonergb/rules.mk:17]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sneakbox/aliceclonergb/readme.md (hunk 4 above)
Comment (SneakboxKB) [reply keyboards/sneakbox/aliceclonergb/readme.md:11]: Done. Added instructions on how to enter bootloader for this firmware.

File: keyboards/sneakbox/aliceclonergb/rules.mk [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+
+# Build Options
+#   change yes to no to disable
+#
+BOOTMAGIC_ENABLE = yes      # Enable Bootmagic Lite
+MOUSEKEY_ENABLE = yes       # Mouse keys
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+NKRO_ENABLE = no            # USB Nkey Rollover
```
Comment (fauxpark) [inline keyboards/sneakbox/aliceclonergb/rules.mk:15]:
```suggestion
NKRO_ENABLE = no            # Enable N-Key Rollover
```

**Review by fauxpark — CHANGES_REQUESTED** [review]

**Comment by SneakboxKB** [comment]
Hello, just following up on this PR. Looks like outstanding issues have been resolved. Please let me know if there's anything else that needs correction. Thanks.

**Review by fauxpark — APPROVED** [review]

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/sneakbox/aliceclonergb/keymaps/via/keymap.c [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+  } else {
+    writePinLow(D7);
+  }
+  if (led_state.caps_lock) {
+    writePinHigh(D6);
+  } else {
+    writePinLow(D6);
+  }
+  return false;
+}
```
[the remark this answers is not in the document]
Comment (SneakboxKB) [reply keyboards/sneakbox/aliceclonergb/keymaps/via/keymap.c]: Thanks. I removed all the encoder and LED calls in keymap.c as these were already moved to aliceclone.c and config.h respectively.

**Review by drashna — APPROVED** [review]

**Pull request merged**